src/main/scala/components/memory/memory.scala

```diff
@@ -53,147 +60,14 @@ class DualPortedCombinMemory(size: Int, memfile: String) extends BaseDualPortedM
- * Input:  sext, true if we should sign extend the result
- *
- * Output: the data read and sign extended
- */
-class DMemIO extends Bundle {
-  val address   = Input(UInt(32.W))
-  val writedata = Input(UInt(32.W))
-  val memread   = Input(Bool())
-  val memwrite  = Input(Bool())
-  val maskmode  = Input(UInt(2.W))
-  val sext      = Input(Bool())
-
-  val readdata  = Output(UInt(32.W))
-}
-
-/**
- * This is the actual memory. You should never directly use this in the CPU.
- * This module should only be instantiated in the Top file.
- *
- * Note: this module is **deprecated**. See [[DualPortedCombinMemory]] for a functionally equivalent backing memory
- * which utilizes the newer memory module system.
- *
- * The I/O for this module is defined in [[IMemIO]] and [[DMemIO]].
- */
-class DualPortedMemory(size: Int, memfile: String) extends Module {
-  val io = IO(new Bundle {
-    val imem = new IMemIO
-    val dmem = new DMemIO
-  })
-  io := DontCare
-
-  val memory = Mem(math.ceil(size.toDouble / 4).toInt, UInt(32.W))
-  loadMemoryFromFile(memory, memfile)
-
-  when(io.imem.address >= size.U) {
-    io.imem.instruction := 0.U
-  }.otherwise {
-    io.imem.instruction := memory(io.imem.address >> 2)
-  }
-
-  when(io.dmem.memread) {
-    assert(io.dmem.address < size.U)
-    val readdata = Wire(UInt(32.W))
-
-    when(io.dmem.maskmode =/= 2.U) { // When not loading a whole word
-      val offset = io.dmem.address(1, 0)
-      readdata := memory(io.dmem.address >> 2) >> (offset * 8.U)
-      when(io.dmem.maskmode === 0.U) { // Reading a byte
-        readdata := memory(io.dmem.address >> 2) & 0xff.U
-      }.otherwise {
-        readdata := memory(io.dmem.address >> 2) & 0xffff.U
-      }
-    }.otherwise {
-      readdata := memory(io.dmem.address >> 2)
-    }
-
-    when(io.dmem.sext) {
-      when(io.dmem.maskmode === 0.U) {
-        io.dmem.readdata := Cat(Fill(24, readdata(7)), readdata(7, 0))
-      }.elsewhen(io.dmem.maskmode === 1.U) {
-        io.dmem.readdata := Cat(Fill(16, readdata(15)), readdata(15, 0))
-      }.otherwise {
-        io.dmem.readdata := readdata
-      }
-    }.otherwise {
-      io.dmem.readdata := readdata
-    }
-  }
-
-  when(io.dmem.memwrite) {
-    assert(io.dmem.address < size.U)
-    when(io.dmem.maskmode =/= 2.U) { // When not storing a whole word
-      val offset = io.dmem.address(1, 0)
-      // first read the data since we are only overwriting part of it
-      val readdata = Wire(UInt(32.W))
-      readdata := memory(io.dmem.address >> 2)
-      // mask off the part we're writing
-      val data = Wire(UInt(32.W))
-      when(io.dmem.maskmode === 0.U) { // Reading a byte
-        data := readdata & ~(0xff.U << (offset * 8.U))
-      }.otherwise {
-        data := readdata & ~(0xffff.U << (offset * 8.U))
-      }
-      memory(io.dmem.address >> 2) := data | (io.dmem.writedata << (offset * 8.U))
-    }.otherwise {
-      memory(io.dmem.address >> 2) := io.dmem.writedata
-    }
-  }
-}
-
+}
```

Comment:
You should set your editor to put newlines at the end of files :)

Reply:
That's strange, it normally does.